socorro/processor/rules/mozilla.py

```diff
@@ -449,6 +449,33 @@ def action(self, raw_crash, dumps, processed_crash, tmpdir, status):
             status.add_note(f"Breadcrumbs: malformed: {jexc.message}")
 
 
+class MacBootArgsRule(Rule):
+    """Extracts mac_boot_args from json_dump
+
+    If there's a mac_boot_args in the json_dump and it's not an empty value, this
+    copies it to the processed_crash.
+
+    """
+
+    def predicate(self, raw_crash, dumps, processed_crash, tmpdir, status):
+        return bool(glom(processed_crash, "json_dump.mac_boot_args", default=None))
+
+    def action(self, raw_crash, dumps, processed_crash, tmpdir, status):
+        value = processed_crash["json_dump"]["mac_boot_args"]
```

Comment:
am I understanding correctly that `processed_crash["json_dump"]["mac_boot_args"]` won't raise KeyError, because predicate ensures that if it didn't exist before it's now None?

Reply:
wait no, it won't throw KeyError because predicate ensures that action isn't called unless there is a truthy value at that path?